dataRetriever/factory/resolverscontainer/baseResolversContainerFactory.go

```diff
@@ -278,7 +284,8 @@ func (brcf *baseResolversContainerFactory) createMiniBlocksResolver(
 func (brcf *baseResolversContainerFactory) generatePeerAuthenticationResolver() error {
 	identifierPeerAuth := common.PeerAuthenticationTopic
 	shardC := brcf.shardCoordinator
-	resolverSender, err := brcf.createOneResolverSender(identifierPeerAuth, EmptyExcludePeersOnTopic, shardC.SelfId())
+	numIntraShardPeers := brcf.numTotalPeers - brcf.numCrossShardPeers
```

Comment:
maybe make `numIntraShardPeers` part of the `baseResolversContainerFactory`? you do the subtraction a lot of times, the same way